pkg/standalone/common.go

```diff
@@ -44,9 +46,64 @@ func binaryFilePath(binaryDir string, binaryFilePrefix string) string {
 }
 
 func DefaultComponentsDirPath() string {
-	return path_filepath.Join(defaultDaprDirPath(), defaultComponentsDirName)
+	return path_filepath.Join(defaultDaprDirPath(), utils.DefaultComponentsDirName)
+}
+
+func DefaultResourcesDirPath() string {
+	return path_filepath.Join(defaultDaprDirPath(), utils.DefaultResourcesDirName)
+}
+
+// DefaultResourcesDirPrecedence returns the path to the resources directory if it exists, or otherwise the components directory.
+// TODO: Remove this function and use `DefaultResourcesDirPath` when `--components-path` flag is removed.
+func DefaultResourcesDirPrecedence() string {
+	defaultResourcesDirPath := DefaultResourcesDirPath()
+	if _, err := os.Stat(defaultResourcesDirPath); os.IsNotExist(err) {
+		return DefaultComponentsDirPath()
+	}
+	return defaultResourcesDirPath
 }
 
 func DefaultConfigFilePath() string {
 	return path_filepath.Join(defaultDaprDirPath(), defaultConfigFileName)
 }
+
+// emptyAndCopyFiles copies files from src to dest. It deletes the existing files in dest before copying from src.
+// TODO: Remove this function when `--components-path` flag is removed.
+func emptyAndCopyFiles(src, dest string) error {
```

Comment:
Can we add unit tests for this?

Reply:
most of the logic like -  "when components dir exist, when component dir does not exist" are covered under E2E tests. So, IMO we can ignore the UT because we can not verify the actual behaviour with UT here and also this function is temporary.